contracts/src/v0.8/automation/dev/v2_3/AutomationRegistry2_3.sol

```diff
@@ -355,7 +355,9 @@ contract AutomationRegistry2_3 is AutomationRegistryBase2_3, OCR2Abstract, Chain
       offchainConfig
     );
 
-    delete s_registrars;
+    for (uint256 idx = 0; idx < s_registrars.length(); idx++) {
+      s_registrars.remove(s_registrars.at(idx));
+    }
```

Comment:
I think this is potentially buggy because `remove()` will adjust the index of items in the set on each delete. I think we need to count backwards instead, like this:

```sol
for (int256 idx = int256(s_registrars.length()); idx > -1; idx--) {
  s_registrars.remove(s_registrars.at(idx));
}
```

Reply:
wow, good catch! 

I just tested in remix, and indeed, using the above code couldnt remove all, it will have leftover. The right way is to cleanup using reverse order, as you recommended.

I added a foundry test to cover this scenario. If you see [this](https://github.com/smartcontractkit/chainlink/pull/12885/commits/ff05527737887a5e9a71fe5e48a118e4c81ceb40) commit , it fails bc of registrar length doesnt match.

with the latest commit, s_registrars is cleaned up properly.